src/main/java/dev/corestone/lotrings/abilities/FortuneAbility.java

```diff
@@ -0,0 +1,50 @@
+package dev.corestone.lotrings.abilities;
+
+import dev.corestone.lotrings.LOTRings;
+import dev.corestone.lotrings.Ring;
+import dev.corestone.lotrings.RingState;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.block.BlockBreakEvent;
+import org.bukkit.inventory.ItemStack;
+
+public class FortuneAbility extends AbilitySuper {
+
+    private float fortuneLevel;
+
+    public FortuneAbility(LOTRings plugin, Ring ring, String abilityName) {
+        super(plugin, ring, abilityName);
+        try {
+            this.fortuneLevel = plugin.getAbilityDataManager().getAbilityFloatData(abilityName, "fortune-level");
+        } catch (Exception e) {
+            sendLoadError();
+        }
+    }
+
+    @Override
+    public void switchState(RingState ringState) {
+        super.switchState(ringState);
+        if (ringState != RingState.HELD) {
+            HandlerList.unregisterAll(this);
```

Comment:
Instead of unregistering, registering, unregistering, and registering events over and over again, just check if the ring is held in your BlockBreakEvent function (just use: " if(!ring.isHeld()) return; ").

Reply:
You have access to all the public classes and objects in the parent AbilitySuper.jar file if you want to take a look. This includes the plugin class, ring, abilityName, and abilityDisplayName.